src/h.js

```diff
@@ -1,3 +1,24 @@
+var nodePool = []
+
+function reuseNode(node, tag, data, children) {
+  node.tag = tag
+  node.data = data
+  node.children = children
+  return node
+}
+
+export function getNode(tag, data, children) {
+  return nodePool.length > 0 
+    ? reuseNode(nodePool.pop(), tag, data, children)
+    : { tag: tag, data: data, children: children }
+}
+
+export function recoverNode(node) {
+  if (typeof node !== "string") {
+    nodePool.push(node)
```

Comment:
Let’s do some quick analysis. Let’s define a `pass` being a `patch` call. At each call we have:
 - `n` nodes that will be created
 - `m` old nodes 
 - `t` dereferenced nodes that will be collected by the GC later
Let’s also say that a GC pass of `t` is linear (`O(t)`).

Here is a comparison of the current implementation, the PR one and the one using a “only-grow stack”:

|              | Avg space complexity | GC time complexity
|--------|-----------------------|--------------------
 Current | `O(n+m+t)`                 | `O(t)`
 PR         | `O(n+max(m, t))`        | `O(0)`
 OGS      | `O(n+max(m, max(t, t-1, … t0)))`        | `O(0)`